### PR TITLE
use main-name for named-amd and global builds

### PR DESCRIPTION
**Pull request by ghedamat**
currently only `cjs` and `amd` builds get the the name from the `main` option

with this PR the same happens for the `globals` and the `named-amd` dists

@rpflorence let me know what you think and thanks a lot for all your work!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -20,9 +20,9 @@ function validateOptions(options) {
   if (!options.global) throw new Error('You must provide a `global` option so I know what to attach your main script to on the window.');
 }
 
-function makeConcatDist(dir) {
+function makeConcatDist(dir, options) {
```
Comment (ghedamat) [inline index.js:23]: I'm not in love with the idea of passing `options` down to `makeConcatDist` and `concat` but with this first pass I tried to change the least amount possible

**Comment by ghedamat** [comment]
also, I'll add test shortly, sry about that :)

**Comment by ghedamat** [comment]
tests have been added, it's not a great testing structure but it should do

**Comment by cibernox** [comment]
I was about to request this right now. I don't want my output files to be named `main.js`.

Is it good to merge?

**Comment by ryanflorence** [comment]
This looks good, I never intended to force "main" on people (hence the main option). Just a bug left over from copy/pasting my original implementation to a general one.

**Comment by ghedamat** [comment]
@rpflorence works for me, let me know if something else is needed, thanks!

**Comment by cibernox** [comment]
I just want to check for updates here. I am eager to see this merged, and apparently everybody agrees with this PR.

**Comment by cibernox** [comment]
Is this waiting for something?

**Pull request closed without merging**